src/client-side-metrics/operation-metrics-collector.ts

```diff
@@ -58,6 +75,33 @@ enum MetricsCollectorState {
   OPERATION_COMPLETE,
 }
 
+// This method swallows errors when metrics debugging is not enabled so
+// that errors don't bubble up to the user.
+function withMetricsDebug<T>(fn: () => T): T | undefined {
+  try {
+    return fn();
+  } catch (e) {
+    if (METRICS_DEBUG) {
+      console.warn(e);
+    }
+  }
+  return;
+}
+
+// Checks that the state transition is valid and if not it throws a warning.
+function checkState<T>(
+  currentState: MetricsCollectorState,
+  allowedStates: MetricsCollectorState[],
+  fn: () => T
+): T | undefined {
+  if (allowedStates.includes(currentState)) {
+    return fn();
+  } else {
+    console.warn('Invalid state transition');
```

Comment:
Did you intend to have this always log, or should this also be behind METRICS_DEBUG?

Reply:
I ended up gating this with an environment variable and actually throwing an error when the variable is true. We have to change the npm script in package json to use the variable in tests:

ie. npx cross-env METRICS_DEBUG=true mocha build/system-test --timeout 600000

Should we use the variable in tests too?